crates/bitwarden/src/platform/fido2/authenticator.rs

```diff
@@ -92,242 +90,373 @@ impl<'a> Fido2Authenticator<'a> {
-                    key_algorithm: "".to_owned().encrypt_with_key(key)?,
-                    key_curve: "".to_owned().encrypt_with_key(key)?,
-                    key_value: "".to_owned().encrypt_with_key(key)?,
-                    rp_id: "".to_owned().encrypt_with_key(key)?,
-                    user_handle: Some("".to_owned().encrypt_with_key(key)?),
-                    user_name: Some("".to_owned().encrypt_with_key(key)?),
-                    counter: "".to_owned().encrypt_with_key(key)?,
-                    rp_name: Some("".to_owned().encrypt_with_key(key)?),
-                    user_display_name: Some("".to_owned().encrypt_with_key(key)?),
-                    discoverable: "".to_owned().encrypt_with_key(key)?,
-                    creation_date: chrono::offset::Utc::now(),
+        // Insert the received UV to be able to return it later in check_user
+        self.selected_uv
+            .lock()
+            .expect("Mutex is not poisoned")
+            .replace(request.options.uv);
+
+        let response = authenticator
+            .get_assertion(ctap2::get_assertion::Request {
+                rp_id: request.rp_id,
+                client_data_hash: request.client_data_hash.into(),
+                allow_list: request
+                    .allow_list
+                    .map(|l| {
+                        l.into_iter()
+                            .map(TryInto::try_into)
+                            .collect::<Result<Vec<_>, _>>()
+                    })
+                    .transpose()?,
+                extensions: request
+                    .extensions
+                    .map(|e| serde_json::from_str(&e))
+                    .transpose()?,
+                options: passkey::types::ctap2::make_credential::Options {
+                    rk: request.options.rk,
+                    up: true,
+                    uv: request.options.uv != UV::Discouraged,
                 },
-            },
+                pin_auth: None,
+                pin_protocol: None,
+            })
+            .await;
+
+        let response = match response {
+            Ok(x) => x,
+            Err(e) => return Err(format!("get_assertion error: {e:?}").into()),
+        };
+
+        let authenticator_data = response.auth_data.to_vec();
+        let credential_id = response
+            .auth_data
+            .attested_credential_data
+            .ok_or("Missing attested_credential_data")?
+            .credential_id()
+            .to_vec();
+
+        Ok(GetAssertionResult {
+            credential_id,
+            authenticator_data,
+            signature: response.signature.into(),
+            user_handle: response.user.ok_or("Missing user")?.id.into(),
+            selected_credential: self.get_selected_credential()?,
         })
     }
 
-    // TODO: Fido2CredentialView contains all the fields, maybe we need a Fido2CredentialListView?
+    // TODO(Fido2): We need some changes in the passkey crate to do this correctly
```

Comment:
Would we need this function now then? It's basically just calling `find_credentials`, so the clients could be doing that directly.

Reply:
The idea was that the SDK would act as the main point of access for `Fido2Credentials`, calling their own `find_credentials` is a bit counter productive in that regard. But this will only really make a difference when the SDK starts having state, in which case you would no longer have access to `find_credentials`

Reply:
That's sounds reasonable yeah! I'll remove the TODO comment for now then.